msteams-platform/concepts/deploy-and-publish/appsource/prepare/frequently-failed-cases.md

```diff
@@ -1,246 +1,322 @@
-* Use the available viewport area efficiently. Do not waste it on using huge logos inside the configuration pop up.
+### 6.2 Bot welcome messages
 
-![OneNote allows users to paste a OneNote link in case notes can not be found](../../../../assets/images/faq/tab-onenote-config.png)
+Bots should almost always send first-run welcome messages. For the best experience, the message should include the value proposition of your bot, how to configure the bot, and briefly describe all supported bot commands. You can display the messaging using an Adaptive Card with buttons for better readability. [How to trigger a bot welcome message](~/bots/how-to/conversations/send-proactive-messages.md).
 
-![Users can always create new plan on planner in case there no existing ones](../../../../assets/images/faq/tab-planner-config.png)
+Channel bot welcome messages are optional during first run, especially if bot is available in personal scope performing similar actions. This bot must not send a personal message to all members of the team or channel as that will be considered as spamming. The welcome message should also mention the person who installed/added the bot to the channel.
 
-![SharePoint also allows user to directly paste a SharePoint link](../../../../assets/images/faq/tab-sp-config.png)
+Notification-only bots must send a welcome message that includes a message conveying, "I am a notification-only bot and will not be able to reply to your chats".
 
-### &#9989; Tabs in channel - Member access
+### 6.3 Bot message spamming
 
-* A tab configured by a member in a channel scope must be accessible to the other members without having to seek permissions from the member who configured the tab.
-* The app must provide the permission management options upfront if the tab is for private or restricted use or requires any permissions from the member who configured the tab.
+Bots must not spam users by sending multiple messages in short succession.
 
-### &#9989; Bots must always be responsive and fail gracefully
+* **Bot channel messages**: Don't spam users by creating separate new chat posts. Create a single thread post with replies in the same thread.
+* **Bot personal messages**: Don't send multiple messages. Send one message with complete information. Repeating the same welcome message over regular intervals is not allowed and is considered spamming.
 
-Your bot must be responsive to any command and not dead-end the user. Here are some tips to help your bot intelligently respond to users:
+### 6.4 Bot notifications
 
-* **Use command lists**. Analyzing user input or predicting user intent is hard. Instead of letting users guess what your bot can do, provide a list of commands your bot understands.
+Bot notifications must include content relevant for the scope you define for the bot (`team`, `groupchat`, or `personal`).
```

Comment:
Is the “must” saying the notification content must be relevant to the context (team, group chat, or. personal)?

Reply:
Yes